vm/devices/net/mana_driver/src/gdma_driver.rs

```diff
@@ -219,21 +216,20 @@ impl<T: DeviceBacking> Drop for GdmaDriver<T> {
         );
 
         const AFTER_DESTROY_HWC: &str = "After GdmaDriver destroy HWC";
-        let header = match self.wait_for_vf_to_own_shmem().context(AFTER_DESTROY_HWC) {
-            Ok(hdr) => hdr,
-            Err(e) => {
-                tracing::error!(context = AFTER_DESTROY_HWC, error = %e);
-                return;
-            }
-        };
+        let header = self.wait_for_vf_to_own_shmem().context(AFTER_DESTROY_HWC);
+        if let Err(e) = header {
+            tracing::error!(context = %e.to_string(), error = ?e.root_cause());
+            return;
+        }
 
+        let header = header.unwrap();
         if !header.is_response() {
             tracing::error!(context = AFTER_DESTROY_HWC, error = "expected response");
```

Comment:
this is not a common routine, so you can just do `tracing::error!("unexpected response for HWC destroy")